src/main/java/com/kainos/ea/daos/RoleDao.java

```diff
@@ -15,49 +16,74 @@
             Statement statement = connection.createStatement();
 
-            ResultSet resultSet = statement.executeQuery(
-                    "SELECT jr.jobRoleId, jr.roleName, jr.location,"
-                            + "b.bandValue AS band,"
-                            + "c.capabilityName AS capability, "
-                            + "jr.closingDate, jr.status "
-                            + "FROM JobRoles jr "
-                            + "JOIN Bands b ON jr.bandId = b.bandId "
-                            + "JOIN Capabilities c ON jr.capabilityId = "
-                            + "c.capabilityId "
-                            + "WHERE jr.status = true");
+            String query = "SELECT jr.jobRoleId, jr.roleName, jr.location, "
+                    + "b.bandValue AS band, "
+                    + "c.capabilityName AS capability, "
+                    + "jr.closingDate, jr.status "
+                    + "FROM JobRoles jr "
+                    + "JOIN Bands b ON jr.bandId = b.bandId "
+                    + "JOIN Capabilities c ON jr.capabilityId = c.capabilityId "
+                    + "WHERE jr.status = true ";
+
+            // Apply ordering if orderBy and direction are provided
+            if (orderBy != null && direction != null) {
+                query += String.format("ORDER BY %s %s", orderBy, direction);
+            }
 
+            ResultSet resultSet = statement.executeQuery(query);
 
             while (resultSet.next()) {
                 RoleResponse roleResponse = new RoleResponse(
                         resultSet.getInt("jobRoleId"),
-                                        resultSet.getString("roleName"),
-                                        Locations.fromString(
-                                                resultSet.getString(
-                                                "location")),
-                                        resultSet.getString("band"),
-                                        resultSet.getString("capability"),
+                        resultSet.getString("roleName"),
+                        Locations.fromString(resultSet.getString("location")),
+                        resultSet.getString("band"),
+                        resultSet.getString("capability"),
                         resultSet.getBoolean("status"),
                         resultSet.getTimestamp("closingDate")
-                                );
+                );
                 roles.add(roleResponse);
             }
             return roles;
         } catch (DatabaseConnectionException e) {
             throw new FailedToRetrieveException(Entity.JOB_ROLE);
+        } catch (SQLException e) {
+            System.err.println("SQLException occurred: " + e.getMessage());
+            throw e;
+        }
+    }
+
+    private static <T extends Enum<T>> boolean isValidEnumValue(
+            final Class<T> enumClass, final String value) {
+        for (T enumValue : enumClass.getEnumConstants()) {
```

Comment:
I expected to see something like `Object.... values()`. A one liner for checking the enum.